diff --git a/node_modules/expo/android/build.gradle b/node_modules/expo/android/build.gradle
index 0bcdc37..a3a770f 100644
--- a/node_modules/expo/android/build.gradle
+++ b/node_modules/expo/android/build.gradle
@@ -53,7 +53,7 @@ buildscript {
     if (ext.has("kotlinVersion")) {
       ext.kotlinVersion()
     } else {
-      ext.safeExtGet("kotlinVersion", "1.6.10")
+      ext.safeExtGet("kotlinVersion", "1.6.20")
     }
   }
 
