diff --git a/node_modules/react-native-pager-view/ios/ReactNativePageView.m b/node_modules/react-native-pager-view/ios/ReactNativePageView.m
index 30e7565..40a98c7 100644
--- a/node_modules/react-native-pager-view/ios/ReactNativePageView.m
+++ b/node_modules/react-native-pager-view/ios/ReactNativePageView.m
@@ -182,6 +182,9 @@ - (void)setReactViewControllers:(NSInteger)index
     uint16_t coalescingKey = _coalescingKey++;
     
     if (animated == YES) {
+        if (controller == [self currentlyDisplayed]) {
+            return;
+        }
         self.animating = YES;
     }
     
