diff --git a/node_modules/react-native-pager-view/ios/ReactNativePageView.m b/node_modules/react-native-pager-view/ios/ReactNativePageView.m
index 30e7565..d4bf972 100644
--- a/node_modules/react-native-pager-view/ios/ReactNativePageView.m
+++ b/node_modules/react-native-pager-view/ios/ReactNativePageView.m
@@ -1,6 +1,6 @@
 
 #import "ReactNativePageView.h"
-#import "React/RCTLog.h"
+#import <React/RCTLog.h>
 #import <React/RCTViewManager.h>
 
 #import "UIViewController+CreateExtension.h"
@@ -175,13 +175,16 @@ - (void)setReactViewControllers:(NSInteger)index
                       direction:(UIPageViewControllerNavigationDirection)direction
                        animated:(BOOL)animated
                        shouldCallOnPageSelected:(BOOL)shouldCallOnPageSelected {
-    if (self.reactPageViewController == nil) {
+    if (self.reactPageViewController == nil || !self.reactPageViewController.viewControllers) {
         return;
     }
     __weak ReactNativePageView *weakSelf = self;
     uint16_t coalescingKey = _coalescingKey++;
     
     if (animated == YES) {
+        if (controller == [self currentlyDisplayed]) {
+            return;
+        }
         self.animating = YES;
     }
     
@@ -337,7 +340,6 @@ - (void)pageViewController:(UIPageViewController *)pageViewController
         didFinishAnimating:(BOOL)finished
    previousViewControllers:(nonnull NSArray<UIViewController *> *)previousViewControllers
        transitionCompleted:(BOOL)completed {
-    
     if (completed) {
         UIViewController* currentVC = [self currentlyDisplayed];
         NSUInteger currentIndex = [self.reactSubviews indexOfObject:currentVC.view];
