diff --git a/node_modules/react-native-nordic-dfu/ios/RNNordicDfu.m b/node_modules/react-native-nordic-dfu/ios/RNNordicDfu.m
index dd68681..9e00fa8 100644
--- a/node_modules/react-native-nordic-dfu/ios/RNNordicDfu.m
+++ b/node_modules/react-native-nordic-dfu/ios/RNNordicDfu.m
@@ -233,7 +233,7 @@ - (void)logWith:(enum LogLevel)level message:(NSString * _Nonnull)message
         initiator.alternativeAdvertisingNameEnabled = alternativeAdvertisingNameEnabled;
 
         // Change for iOS 13
-        initiator.packetReceiptNotificationParameter = 1; //Rate limit the DFU using PRN.
+        // initiator.packetReceiptNotificationParameter = 1; //Rate limit the DFU using PRN.
         [NSThread sleepForTimeInterval: 2]; //Work around for being stuck in iOS 13
         // End change for iOS 13
 
