diff --git a/node_modules/react-native-pager-view/ios/ReactNativePageView.m b/node_modules/react-native-pager-view/ios/ReactNativePageView.m
index cf2bd57..c37a09e 100644
--- a/node_modules/react-native-pager-view/ios/ReactNativePageView.m
+++ b/node_modules/react-native-pager-view/ios/ReactNativePageView.m
@@ -1,6 +1,6 @@
 
 #import "ReactNativePageView.h"
-#import "React/RCTLog.h"
+#import <React/RCTLog.h>
 #import <React/RCTViewManager.h>
 
 #import "UIViewController+CreateExtension.h"
@@ -167,6 +167,9 @@ - (void)setReactViewControllers:(NSInteger)index
     uint16_t coalescingKey = _coalescingKey++;
     
     if (animated == YES) {
+        if (controller == [self currentlyDisplayed]) {
+            return;
+        }
         self.animating = YES;
     }
     
